aptos-move/framework/aptos-framework/sources/fungible_asset.move

```diff
@@ -107,57 +147,75 @@ module aptos_framework::fungible_asset {
                 decimals,
             }
         );
+        move_to(metadata_object_signer,
+            FungibleAssetMetadataEvents {
+                mint_events: object::new_event_handle<MintEvent>(metadata_object_signer),
+                burn_events: object::new_event_handle<BurnEvent>(metadata_object_signer),
+            }
+        );
         let metadata = object::object_from_constructor_ref<FungibleAssetMetadata>(constructor_ref);
         (MintRef { metadata }, TransferRef { metadata }, BurnRef { metadata })
     }
 
+    #[view]
     /// Get the current supply from `metadata`.
-    public fun supply<T: key>(metadata: &Object<T>): u64 acquires FungibleAssetMetadata {
-        borrow_fungible_metadata(metadata).supply
+    public fun supply<T: key>(metadata: Object<T>): u64 acquires FungibleAssetMetadata {
+        borrow_fungible_metadata(&metadata).supply
     }
 
+    #[view]
     /// Get the maximum supply from `metadata`.
-    public fun maximum<T: key>(metadata: &Object<T>): Option<u64> acquires FungibleAssetMetadata {
-        borrow_fungible_metadata(metadata).maximum
+    public fun maximum<T: key>(metadata: Object<T>): Option<u64> acquires FungibleAssetMetadata {
```

Comment:
Ahh... so this is still up for discussion.

considering that we are doing a copy on all these functions anyway (we're passing in the value), it might not be the end of the world to expect that this is coerced to the appropriate type. Perhaps this also creates the opportunity for someone to validate it is of a given type before coercing. Let them have their own error handling if they want it.

Note, we will still need to re-validate `exists<T>(object_address(obj))` since an `Object<T>` just implies it existed at some point.

Reply:
Yeah I was thinking exactly this too. We need to think about this a bit more about how to do this though